### PR TITLE
Feat/425 publish dataset and collection

### LINKED ISSUE #425: Publish Dataset:  Publish Dataset and Dataset Collection at the same time
In the JSF, when you click 'Publish' on a Dataset that belongs to an unpublished Collection, the Popup gives the options of publishing both.

Screenshot 2024-07-01 at 9 26 04 AM

To implement in the SPA, we need the Publish Collection useCase available.

### PR DESCRIPTION
**Pull request by ekraffmiller**
## What this PR does / why we need it:
Implements the Publish Dataset feature where both the dataset and it's parent collection are not published. In this case, the Publish Dataset Modal allows the user to publish both the dataset and collection at the same time.

**Note** that this feature depends on a js-dataverse PR that is not yet merged: https://github.com/IQSS/dataverse-client-javascript/pull/199.  When it is merged, package.json can be updated to use the latest 'alpha' version.

## Which issue(s) this PR closes:

- Closes #425 

## Special notes for your reviewer:
I'm using` feat/461-publish-collection` as the base branch because it contains the integration code to call the Publish Collection js-dataverse use case.

## Suggestions on how to test this:
**Step 1: Run the Development Environment**
1. Execute `npm i`.
2. Navigate with `cd packages/design-system && npm run build`.
3. Return with `cd ../../`.
4. Ensure you have a `.env` file similar to `.env.example`, with the variable `VITE_DATAVERSE_BACKEND_URL=http://localhost:8000`.
5. Navigate with `cd dev-env`.
6. Start the environment using `./run-env.sh unstable`.
7. To verify the environment, visit <http://localhost:8000> and check your local Dataverse installation.

**Step 2: Test the feature**
1. Create a collection. When it is created, it will have the 'Unreleased' status
2. Within the new collection, create a dataset. 
3. On the Dataset page,  Choose Publish Dataset,  and Publish.
4. On the Publish Dataset Modal, there will be a text asking if you want to publish the parent collection as well as the dataset.  It will contain a link to then parent collection
5. Choose 'Continue' to publish the dataset and collection
6. When the publish is completed, check that the dataset and the collection are both published.  The dataset should be Version 1.0, and the collection should no longer have the 'Unreleased' status

## Does this PR introduce a user interface change? If mockups are available, please link/include them here:

This is not a change from the JSF, but here is what the JSF modal looks like.  'Unpublished' is the name of the parent collection.
Screenshot 2024-10-06 at 12 09 15 PM

### CODE REVIEW

**Comment by g-saracca** [comment]
Hi @ekraffmiller could you update js-dataverse package version with `2.0.0-alpha.2`, https://github.com/IQSS/dataverse-client-javascript/pull/199 is already merged, thanks!

**Review by g-saracca — CHANGES_REQUESTED** [review]
@ekraffmiller , looks awesome! just some minor comments and there is a conflict in package.json

File: tests/component/sections/dataset/dataset-publish/PublishDatasetModal.spec.tsx [hunk 1]
```diff
@@ -1,22 +1,32 @@
...
+import { UpwardHierarchyNodeMother } from '../../../shared/hierarchy/domain/models/UpwardHierarchyNodeMother'
 
 describe('PublishDatasetModal', () => {
-  it('does not render the radio buttons when releasedVersionExists is false', () => {
+  it('display modal for never released dataset', () => {
     const handleClose = cy.stub()
     const repository = {} as DatasetRepository // Mock the repository as needed
     repository.publish = cy.stub().as('repositoryPublish').resolves()
+    const collectionRepository = {} as CollectionRepository
+    collectionRepository.publish = cy.stub().as('collectionRepositoryPublish').resolves()
```
[the remark this answers is not in the document]
Comment (ekraffmiller) [reply tests/component/sections/dataset/dataset-publish/PublishDatasetModal.spec.tsx:13]: Good point, I will add to test that it is called

**Review by g-saracca — APPROVED** [review]
Looks great! thanks, approving!

**Comment by ofahimIQSS** [comment]
tested in local - no issues found. 
[Testing of 520.docx](https://github.com/user-attachments/files/17398519/Testing.of.520.docx)

**Pull request merged**